server/api.go

```diff
@@ -228,3 +228,42 @@ func (d dexAPI) ListRefresh(ctx context.Context, req *api.ListRefreshReq) (*api.
 		RefreshTokens: refreshTokenRefs,
 	}, nil
 }
+
+func (d dexAPI) RevokeRefresh(ctx context.Context, req *api.RevokeRefreshReq) (*api.RevokeRefreshResp, error) {
+	id := new(internal.IDTokenSubject)
+	if err := internal.Unmarshal(req.UserId, id); err != nil {
+		d.logger.Errorf("api: failed to unmarshal ID Token subject: %v", err)
+		return nil, err
+	}
+
+	offlineSessions, err := d.s.GetOfflineSessions(id.UserId, id.ConnId)
+	if err != nil {
+		d.logger.Errorf("api: failed to get refresh tokens: %v", err)
+		return nil, err
+	}
+
+	refreshToken, ok := offlineSessions.Refresh[req.ClientId]
+	if !ok {
+		d.logger.Errorf("api: user does not have a refresh token for the client = %s", req.ClientId)
+		return nil, fmt.Errorf("user does not have a refresh token for the client = %s", req.ClientId)
+	}
+
+	// Remove entry from Refresh list of the OfflineSession object.
+	delete(offlineSessions.Refresh, req.ClientId)
+
+	// If the user has no more refresh tokens, the OfflineSession object should be cleaned up.
+	if len(offlineSessions.Refresh) == 0 {
+		if err := d.s.DeleteOfflineSessions(id.UserId, id.ConnId); err != nil {
```

Comment:
Discussed offline and decided to keep the Offline Session object around even if the user does not have active refresh tokens. Might be helpful later on for auditing purposes.